File: app/src/main/java/com/yashovardhan99/healersdiary/fragments/TimePickerFragment.java
package com.yashovardhan99.healersdiary.fragments;

import android.app.Dialog;
import android.app.TimePickerDialog;
import android.content.Context;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import android.text.format.DateFormat;
import android.widget.TimePicker;

import java.util.Calendar;

/**
 * Created by Yashovardhan99 on 1/11/18 as a part of HealersDiary.
 */
public class TimePickerFragment extends DialogFragment implements TimePickerDialog.OnTimeSetListener {
    int hour, minute;
    public TimePickerListener listener;

    public TimePickerFragment(){
        Calendar c = Calendar.getInstance();
        setTime(c);
    }

    private void setTime(Calendar c) {
        hour = c.get(Calendar.HOUR_OF_DAY);
        minute = c.get(Calendar.MINUTE);
    }

    public interface TimePickerListener{
        void onTimeSet(DialogFragment dialogFragment);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try {
            listener = (TimePickerListener) getTargetFragment();
        } catch (ClassCastException e){
            throw new ClassCastException(getTargetFragment() + "must implement TimePickerListener");
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        if(getArguments()!=null){
            Calendar c = Calendar.getInstance();
            c.setTimeInMillis(getArguments().getLong("DATE"));
            setTime(c);
        }
        return new TimePickerDialog(getActivity(), this, hour, minute, DateFormat.is24HourFormat(getContext()));
    }

    @Override
    public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
        this.hour = hourOfDay;
        this.minute = minute;
        listener.onTimeSet(this);
    }
}

File: app/src/main/java/com/yashovardhan99/healersdiary/fragments/TimePickerFragment.kt
package com.yashovardhan99.healersdiary.fragments

import android.app.Dialog
import android.app.TimePickerDialog
import android.content.Context
import android.os.Bundle
import android.text.format.DateFormat
import android.widget.TimePicker
import androidx.fragment.app.DialogFragment
import java.util.*

/**
 * Created by Yashovardhan99 on 1/11/18 as a part of HealersDiary.
 */
class TimePickerFragment : DialogFragment(), TimePickerDialog.OnTimeSetListener {
    internal var hour: Int = 0
    internal var minute: Int = 0
    lateinit var listener: TimePickerListener

    init {
        val c = Calendar.getInstance()
        setTime(c)
    }

    private fun setTime(c: Calendar) {
        hour = c.get(Calendar.HOUR_OF_DAY)
        minute = c.get(Calendar.MINUTE)
    }

    interface TimePickerListener {
        fun onTimeSet(dialogFragment: DialogFragment)
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        try {
            listener = targetFragment as TimePickerListener
        } catch (e: ClassCastException) {
            throw ClassCastException(targetFragment!!.toString() + "must implement TimePickerListener")
        }

    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        if (arguments != null) {
            val c = Calendar.getInstance()
            c.timeInMillis = arguments!!.getLong("DATE")
            setTime(c)
        }
        return TimePickerDialog(activity, this, hour, minute, DateFormat.is24HourFormat(context))
    }

    override fun onTimeSet(view: TimePicker, hourOfDay: Int, minute: Int) {
        this.hour = hourOfDay
        this.minute = minute
        listener.onTimeSet(this)
    }
}
